requirements.txt

```
@@ -0,0 +1,4 @@
+Django>=2.2.2
```

These dependencies are no longer locked. One of these downstream dependencies May release a breaking change/update which will need remediation on our end. 

Pipenv does a great job at locking by default (and checksumming to ensure reproducible environments). A requirements based approach should lock major versions

```suggestion
Django>=2.2.2,<2.3
```

---

What about having two requirements files that emulate the same behavior? For e.g.,

`requirements.txt`
```
django>=2.1.9
gunicorn
```

`pip freeze > requirements_freeze.txt`
```
Django==2.2.2
gunicorn==19.9.0
pytz==2019.1
sqlparse==0.3.0
```

When there is a need to upgrade packages, we can just do `pip install --upgrade -r requirements.txt && pip freeze > requirements_freeze.txt`.

I'd really like to avoid the need of using an extra tool to manage packages. :rofl:

---

This is totally a feasible strategy, and I have done this in the past. However the significant downside is that you (and other developers) need to remember to keep these two files in sync. Additionally, there is no way to specify "dev" developments, like linting or testing tooling.

Pipenv does a great job at that, and solves these problems.

The problems I understand you're trying to solve:

1. Additional tooling required to install dependencies
2. Dependency locking for pipenv is too slow
3. `pipenv run python manage.py` every time is a bit too cumbersome.

I believe whilst these are issues, I don't think they are as major as you make them out to be;

> 1. Additional tooling required to install dependencies

Yes indeed an additional tool is required, but it's no more difficult for the developer to install `pipenv` via `pip` as it is to install `python` in the first place. As a bonus, most python dev's will probably already have `pipenv` on their environment ready to roll - I know I do for a handful of projects far and wide.

> 2. Dependency locking for pipenv is too slow

Totally agree with you on this, but how likely is it that you'll be, a) Adding new dependencies, b) upgrading dependencies? Once the project is built and put into "maintenance" mode, I think this will be a minor concern.

> 3. `pipenv run python manage.py` every time is a bit too cumbersome.

As mentioned above, can be overcome using `pipenv shell`.

---

Righto. Seems like `pipenv` is a better tool in managing deps and migrating back to `pip` is quite unnecessary.